integration/tests/access/cohort1/access_api_test.go

```diff
@@ -585,3 +591,528 @@ func notOutOfRangeError(err error) bool {
+		SetProposalKey(payer, 0, serviceClient.GetAndIncrementSeqNumber()).
+		SetPayer(payer)
+
+	tx, err = serviceClient.SignTransaction(tx)
+	s.Require().NoError(err)
+
+	// Convert the transaction to a message format expected by the access API
+	authorizers := make([][]byte, len(tx.Authorizers))
+	for i, auth := range tx.Authorizers {
+		authorizers[i] = auth.Bytes()
+	}
+
+	// Helper function to convert signatures with ExtensionData
+	convertToMessageSigWithExtensionData := func(sigs []sdk.TransactionSignature, extensionData []byte) []*entities.Transaction_Signature {
+		msgSigs := make([]*entities.Transaction_Signature, len(sigs))
+		for i, sig := range sigs {
+			msgSigs[i] = &entities.Transaction_Signature{
+				Address:       sig.Address.Bytes(),
+				KeyId:         uint32(sig.KeyIndex),
+				Signature:     sig.Signature,
+				ExtensionData: extensionData,
+			}
+		}
+		return msgSigs
+	}
+
+	s.Require().NoError(err)
+
+	// Test WebAuthn extension data with different scenarios
+	testCases := []struct {
+		name          string
+		extensionData []byte
+		description   string
+		expectSuccess bool
+	}{
+		{
+			name:          "webauthn_valid",
+			extensionData: nil,
+			description:   "WebAuthn scheme with minimal extension data",
+			expectSuccess: true,
+		},
+		{
+			name:          "webauthn_invalid_minimal",
+			extensionData: []byte{0x1}, // WebAuthn scheme identifier only
+			description:   "WebAuthn scheme with minimal extension data",
+			expectSuccess: false, // Should fail validation due to incomplete WebAuthn data
+		},
+		{
+			name:          "webauthn_invalid_scheme",
+			extensionData: []byte{0x3, 0x01, 0x02, 0x03}, // Invalid scheme identifier
+			description:   "Invalid authentication scheme",
+			expectSuccess: false,
+		},
+		{
+			name:          "webauthn_malformed_data",
+			extensionData: []byte{0x1, 0x01, 0x02}, // WebAuthn scheme with malformed data
+			description:   "WebAuthn scheme with malformed extension data",
+			expectSuccess: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			s.T().Logf("Testing: %s", tc.description)
+
+			transactionMsg := &entities.Transaction{
+				Script:           tx.Script,
+				Arguments:        tx.Arguments,
+				ReferenceBlockId: tx.ReferenceBlockID.Bytes(),
+				GasLimit:         tx.GasLimit,
+				ProposalKey: &entities.Transaction_ProposalKey{
+					Address:        tx.ProposalKey.Address.Bytes(),
+					KeyId:          uint32(tx.ProposalKey.KeyIndex),
+					SequenceNumber: tx.ProposalKey.SequenceNumber,
+				},
+				Payer:              tx.Payer.Bytes(),
+				Authorizers:        authorizers,
+				PayloadSignatures:  convertToMessageSigWithExtensionData(tx.PayloadSignatures, nil),
+				EnvelopeSignatures: convertToMessageSigWithExtensionData(tx.EnvelopeSignatures, tc.extensionData),
+			}
+
+			if tc.name == "webauthn_valid" {
+				tc.extensionData = s.validWebAuthnExtensionData(transactionMsg)
+			}
```

Comment:
It seems like this if statement does not affect the test itself. The test submits `transactionMsg` which still uses a `nil` extension data. `tc.extensionData` isn't used beyond this point right?
I think the test passes because `nil` is a valid extension data.